service/service.go

```diff
@@ -0,0 +1,151 @@
+	}
+
+	// Get healthcheck with checkers
+	svc.HealthCheck, err = serviceList.GetHealthCheck(cfg, buildTime, gitCommit, version)
+	if err != nil {
+		log.Event(ctx, "failed to create health check", log.FATAL, log.Error(err))
+		return nil, err
+	}
+	if err := svc.registerCheckers(ctx, cfg); err != nil {
+		return nil, errors.Wrap(err, "unable to register checkers")
+	}
+	svc.clients.HealthcheckHandler = svc.HealthCheck.Handler
+
+	// Initialise router
+	r := mux.NewRouter()
+	routes.Init(ctx, r, cfg, svc.clients)
+	m := createMiddleware(cfg)
+	svc.Server = serviceList.GetHTTPServer(cfg.BindAddr, m.Then(r))
+
+	// Start Healthcheck and HTTP Server
+	log.Event(ctx, "service listening...", log.INFO, log.Data{
+		"bind_address": cfg.BindAddr,
+	})
+	svc.HealthCheck.Start(ctx)
+	go func() {
+		if err := svc.Server.ListenAndServe(); err != nil {
+			svcErrors <- errors.Wrap(err, "failure in http listen and serve")
+		}
+	}()
+
+	return svc, nil
+}
+
+// CreateMiddleware creates an Alice middleware chain of handlers
+// to forward collectionID from cookie from header
+func createMiddleware(cfg *config.Config) alice.Chain {
+	return alice.New(dphandlers.CheckCookie(dphandlers.CollectionID))
+}
+
+// Close gracefully shuts the service down in the required order, with timeout
+func (svc *Service) Close(ctx context.Context) error {
+	timeout := svc.Config.GracefulShutdownTimeout
+	log.Event(ctx, "commencing graceful shutdown", log.INFO, log.Data{"graceful_shutdown_timeout": timeout})
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	hasShutdownError := false
+
+	go func() {
+		defer cancel()
+
+		// stop healthcheck, as it depends on everything else
+		if svc.ServiceList.HealthCheck {
+			log.Event(ctx, "stop health checkers", log.INFO)
+			svc.HealthCheck.Stop()
+		}
+
+		// stop any incoming requests
+		if err := svc.Server.Shutdown(ctx); err != nil {
+			log.Event(ctx, "failed to shutdown http server", log.Error(err), log.ERROR)
+			hasShutdownError = true
+		}
+	}()
+
+	// wait for shutdown success (via cancel) or failure (timeout)
+	<-ctx.Done()
+
+	// timeout expired
+	if ctx.Err() == context.DeadlineExceeded {
+		log.Event(ctx, "shutdown timed out", log.ERROR, log.Error(ctx.Err()))
+		return ctx.Err()
+	}
+
+	// other error
+	if hasShutdownError {
+		err := errors.New("failed to shutdown gracefully")
+		log.Event(ctx, "failed to shutdown gracefully ", log.ERROR, log.Error(err))
+		return err
+	}
+
+	log.Event(ctx, "graceful shutdown was successful", log.INFO)
+	return nil
+}
+
+func (svc *Service) registerCheckers(ctx context.Context, cfg *config.Config) (err error) {
+
+	hasErrors := false
+
+	if err = svc.HealthCheck.AddCheck("frontend renderer", svc.clients.Renderer.Checker); err != nil {
+		hasErrors = true
+		log.Event(ctx, "failed to add frontend renderer checker", log.ERROR, log.Error(err))
+	}
+
+	if err = svc.HealthCheck.AddCheck("API Router", svc.routerHealthClient.Checker); err != nil {
```

Comment:
Consider capitalisation of description (ie. `API Router` vs `frontend renderer` above)